src/MArray.jl

```diff
@@ -134,7 +134,7 @@ macro MArray(ex)
             s2s = map(i -> ((isa(ex.args[i], Expr) && ex.args[i].head == :row) ? length(ex.args[i].args) : 1), 1:s1)
             s2 = minimum(s2s)
             if maximum(s2s) != s2
-                error("Rows must be of matching lengths")
+                throw(DimensionMismatch("Rows must be of matching lengths"))
```

Comment:
I feel like the malformed AST which would be passed in this case is more an `ArgumentError` to the macro than a `DimensionMismatch`.  The same could be said of the other macros.